src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,48 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+  health = 100;
+
+  constructor(name) {
+    this.name = name;
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  hidden = false;
+
+  constructor(name) {
+    super(name);
+
+    Animal.alive.push(this);
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  constructor(name) {
+    super(name);
+
+    Animal.alive.push(this);
+  }
+
+  bite(bittenAnimal) {
+    if (bittenAnimal instanceof Carnivore || bittenAnimal.hidden === true) {
+      return;
+    }
+
+    bittenAnimal.health -= 50;
+
+    if (bittenAnimal.health <= 0) {
+      const indexOfBittenAnimal = Animal.alive.indexOf(bittenAnimal);
+
+      Animal.alive.splice(indexOfBittenAnimal, 1);
```

Comment:
To improve code performance and avoid nested loops (indexOf + splice), consider using the filter method to remove the dead animal from the Animal.alive array.